modules/vmfcore/include/vmf/types.hpp

```diff
@@ -103,86 +104,45 @@ namespace vmf
     };
 
     /*!
-    * \brief binary buffer with arbitrary content.
+    * \brief binary buffer with arbitrary content
     */
-    struct vmf_rawbuffer
+    struct vmf_rawbuffer : public std::vector<char>
     {
-        vmf_rawbuffer() : data(nullptr), size(0) {}
+    public:
+        vmf_rawbuffer() : std::vector<char>()
+        { }
 
-        vmf_rawbuffer(const char* _data, const size_t& _size) : size(_size)
-        {
-            if(_data && _size > 0)
-            {
-                data = std::unique_ptr<char[]>(new char[size]);
-                memcpy(data.get(), _data, _size);
-            }
-            else if(!_data && _size == 0)
-                data = nullptr;
-            else
-                VMF_EXCEPTION(IncorrectParamException, "Raw buffer object can't be created. Empty data or invalid size.");
-        }
-
-        vmf_rawbuffer(const vmf_rawbuffer& other) : size(other.size)
-        {
-            if(other.size > 0)
-            {
-                data = std::unique_ptr<char[]>(new char[other.size]);
-                memcpy(data.get(), other.data.get(), other.size);
-            }
-            else
-                data = nullptr;
-        }
+        explicit vmf_rawbuffer(const vmf_rawbuffer& other) : std::vector<char>(other)
+        { }
 
         vmf_rawbuffer(vmf_rawbuffer&& other)
         {
             *this = std::move(other);
         }
 
-        ~vmf_rawbuffer()
+        vmf_rawbuffer& operator=(const vmf_rawbuffer& other)
```

Comment:
aren't you OK with the default compiler-generated code?

BTW, do you miss move/rvalue version?